prog/main.go

```diff
@@ -134,6 +135,53 @@ type appFlags struct {
 	consulInf       string
 }
 
+type filterFlags []string
+
+func (i *filterFlags) String() string {
+	var s string
+	for _, x := range *i {
+		s = x
+	}
+	return s
+}
+
+func (i *filterFlags) Set(value string) error {
+	*i = append(*i, value)
+
+	//fmt.Printf("%q\n",value)
+	u := fmt.Sprintf("%q", value)
+	u = u[1 : len(u)-1]
+
+	soloColonFound := false
+	var title string
+	var label string
+	for k := 1; k < len(u); k++ {
+		if u[k] == ':' {
+			if u[k-1] != '\\' {
+				if soloColonFound == true {
+					fmt.Println("There are multiple unescaped colons")
+					return nil
+				}
+				soloColonFound = true
+				title = u[0:k]
+				label = u[k+1 : len(u)]
+			}
+		}
+	}
+
+	title = strings.Replace(title, "\\:", ":", -1)
+	title = strings.Replace(title, "\\", "", -1)
+	label = strings.Replace(label, "\\:", ":", -1)
+	label = strings.Replace(label, "\\", "", -1)
+
+	v := app.CreateFilterOption(fmt.Sprintf("cmdlinefilter%d", len(*i)), title, render.HasLabel(label))
+	app.ContainerOpts = append(app.ContainerOpts, v)
```

Comment:
> ultimately I do use topologyRegistry's add function inside of the RefreshTopologyOptions() function in api_topologies.go

I still don't understand why you need to build the full options from main. Why not build the options like it was done before (just in init) and simply modify the container options through a topology registry function built specifically for that (which you can also call from main)?